src/Middleware/HttpLogging/src/HttpLoggingInterceptorContext.cs

```diff
@@ -0,0 +1,149 @@
+/// to <see cref="IHttpLoggingInterceptor.OnResponseAsync(HttpLoggingInterceptorContext)"/> except the <see cref="Parameters"/>
+/// which are cleared after logging the request.
+/// </remarks>
+public sealed class HttpLoggingInterceptorContext
+{
+    private HttpContext? _httpContext;
+
+    /// <summary>
+    /// The request context.
+    /// </summary>
+    /// <remarks>
+    /// This property should not be set by user code except for testing purposes.
+    /// </remarks>
+    public HttpContext HttpContext
+    {
+        get => _httpContext ?? throw new InvalidOperationException("HttpContext was not initialized.");
+        // Public for 3rd party testing of interceptors.
+        // We'd make this a required constructor/init parameter but ObjectPool requires a parameterless constructor.
+        set => _httpContext = value ?? throw new ArgumentNullException(nameof(value));
+    }
+
+    /// <summary>
+    /// Gets or sets which parts of the request and response to log.
+    /// </summary>
+    /// <remarks>
+    /// This is pre-populated with the value from <see cref="HttpLoggingOptions.LoggingFields"/>,
+    /// <see cref="HttpLoggingAttribute.LoggingFields"/>, or
+    /// <see cref="HttpLoggingEndpointConventionBuilderExtensions.WithHttpLogging{TBuilder}(TBuilder, HttpLoggingFields, int?, int?)"/>.
+    /// </remarks>
+    public HttpLoggingFields LoggingFields { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of bytes of the request body to log.
+    /// </summary>
+    /// <remarks>
+    /// This is pre-populated with the value from <see cref="HttpLoggingOptions.RequestBodyLogLimit"/>,
+    /// <see cref="HttpLoggingAttribute.RequestBodyLogLimit"/>, or
+    /// <see cref="HttpLoggingEndpointConventionBuilderExtensions.WithHttpLogging{TBuilder}(TBuilder, HttpLoggingFields, int?, int?)"/>.
+    /// </remarks>
+    public int RequestBodyLogLimit { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of bytes of the response body to log.
+    /// </summary>
+    /// <remarks>
+    /// This is pre-populated with the value from <see cref="HttpLoggingOptions.ResponseBodyLogLimit"/>,
+    /// <see cref="HttpLoggingAttribute.ResponseBodyLogLimit"/>, or
+    /// <see cref="HttpLoggingEndpointConventionBuilderExtensions.WithHttpLogging{TBuilder}(TBuilder, HttpLoggingFields, int?, int?)"/>.
+    /// </remarks>
+    public int ResponseBodyLogLimit { get; set; }
+
+    internal long StartTimestamp { get; set; }
+    internal TimeProvider TimeProvider { get; set; } = null!;
+    internal List<KeyValuePair<string, object?>> InternalParameters { get; } = new();
+
+    /// <summary>
+    /// Gets a list of parameters that will be logged as part of the request or response. Values specified in <see cref="LoggingFields"/>
+    /// will be added automatically after all interceptors run. These values are cleared after logging the request.
```

Comment:
> Values specified in <see cref="LoggingFields"/> will be added automatically after all interceptors run. These values are cleared after logging the request.

This makes it sound like only the parameters added automatically after the interceptor are cleared, not all parameters.